src/Collapsible/CollapsibleBody.jsx

```diff
@@ -1,26 +1,28 @@
 import React, { useContext } from 'react';
 import PropTypes from 'prop-types';
 
+import Collapse from '../Collapse';
 import { CollapsibleContext } from './CollapsibleAdvanced';
 import TransitionReplace from '../TransitionReplace';
 
 function CollapsibleBody({
   children, transitionWrapper, tag, ...props
 }) {
-  const { isOpen } = useContext(CollapsibleContext);
+  const { isOpen, unmountOnExit } = useContext(CollapsibleContext);
 
   // Keys are added to these elements so that TransitionReplace
   // will recognize them as unique components and perform the
   // transition properly.
-  const content = isOpen
-    ? React.createElement(tag, { key: 'body', ...props }, children)
-    : <div key="empty" />;
+  const content = React.createElement(tag, { key: 'body', ...props }, children);
+  const transitionBody = isOpen ? content : <div key="empty" />;
 
   if (transitionWrapper) {
-    return React.cloneElement(transitionWrapper, {}, content);
+    return React.cloneElement(transitionWrapper, {}, transitionBody);
   }
   /* istanbul ignore next */
-  return <TransitionReplace>{content}</TransitionReplace>;
+  return unmountOnExit
+    ? <TransitionReplace>{transitionBody}</TransitionReplace>
+    : <Collapse in={isOpen}>{content}</Collapse>;
```

Comment:
It does look like the animation is slightly different between `TransitionReplace` and `Collapse`, in that `TransitionReplace` has a fade-in and fade-out effect in addition to the slide up/down. I'm a bit hesitant to introduce discrepancy in the animation behavior between these 2 components based on a `unmountOnExit` prop.

Also, it does look like `TransitionReplace` passes an `unmountOnExit` prop to `CSSTransition` all the time. Rather than using `Collapse` here, could we instead pass-thru the new `unmountOnExit` prop to `TransitionReplace` to override its default behavior?